### PR TITLE
fix: base API models on the Bitwarden OpenAPI schema

### PR DESCRIPTION
**Pull request by Dubzer**

### CODE REVIEW

File: src/bitwarden/auth.rs [hunk 1]
```diff
@@ -1,18 +1,32 @@
...
+pub enum KdfType {
+    Pbkdf2Sha256 = 0,
+    Argon2id = 1,
+}
+
+impl Default for KdfType {
+    fn default() -> Self {
+        KdfType::Pbkdf2Sha256
+    }
+}
```
Comment (teidesu) [inline src/bitwarden/auth.rs]: is this needed? i see that it's used by `derive(Default)` in `PreloginResponseModel` but im not sure if that's even needed

File: src/bitwarden/auth.rs [hunk 2]
```diff
@@ -1,18 +1,32 @@
...
+    #[serde(rename = "kdf")]
+    pub kdf: KdfType,
+    #[serde(rename = "kdfIterations")]
     pub kdf_iterations: u32,
-    #[serde(rename = "KdfMemory")]
+    #[serde(rename = "kdfMemory")]
     pub kdf_memory: Option<u32>,
-    #[serde(rename = "KdfParallelism")]
+    #[serde(rename = "kdfParallelism")]
     pub kdf_parallelism: Option<u32>,
```
Comment (teidesu) [inline src/bitwarden/auth.rs:30]: https://serde.rs/attr-rename.html

**Review by teidesu — COMMENTED** [review]
overall - it would be nice to have the way you generated the models from the schema somewhere in the comments

tested the thing, it seems to work with older vaultwarden versions too so that's cool ig

File: src/bitwarden/auth.rs (hunk 1 above)
Comment (Dubzer) [reply src/bitwarden/auth.rs]: it isn't. fixed

**Pull request merged**